python_src/src/lamp_py/performance_manager/l0_gtfs_rt_events.py

```diff
@@ -379,7 +372,17 @@ def process_gtfs_rt_files(db_manager: DatabaseManager) -> None:
                 tu_events = process_tu_files(files["tu_paths"], db_manager)
                 events = combine_events(vp_events, tu_events)
 
-            upload_to_database(events, db_manager)
+            # continue events processing if records exist
+            if events.shape[0] > 0:
```

Comment:
`upload_to_database`, `process_trips` and `process_metrics` should only be run if events exist in dataframe. We could throw logging statements to indicate this??